### PR TITLE
feat(MessageComposeView): User can select identity

### PR DESCRIPTION
**Pull request by valentinperignon**

### CODE REVIEW

File: Mail/Views/New Message/Header Cells/ComposeMessageSenderMenu.swift [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    var body: some View {
+        if autocompletionType == nil {
+            VStack(spacing: 0) {
+                HStack {
+                    Text(type.title)
+                        .textStyle(.bodySecondary)
+
+                    if let currentSignature {
+                        Menu {
```
Comment (adrien-coye) [inline Mail/Views/New Message/Header Cells/ComposeMessageSenderMenu.swift:47]: Can we subdivide in a Menu view for eg. ?

File: Mail/Views/New Message/Header Cells/ComposeMessageSenderMenu.swift [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                            ForEach(signatures) { signature in
+                                Button {
```
Comment (adrien-coye) [inline Mail/Views/New Message/Header Cells/ComposeMessageSenderMenu.swift:49]: CellMenuView maybe?

File: MailCore/Cache/DraftContentManager.swift [hunk 3]
```diff
@@ -46,6 +47,34 @@ public class DraftContentManager: ObservableObject {
...
+        let realm = mailboxManager.getRealm()
+        guard let liveIncompleteDraft = realm.object(ofType: Draft.self, forPrimaryKey: incompleteDraft.localUUID) else {
+            return
+        }
+
+        do {
+            let parsedBody = try SwiftSoup.parse(liveIncompleteDraft.body)
+            // If we find the previous signature, we replace it with the new one
+            // otherwise we append the signature at the end of the document
+            if let foundSignatureDiv = try parsedBody.select(".\(Constants.signatureWrapperIdentifier)").first {
```
Comment (adrien-coye) [inline MailCore/Cache/DraftContentManager.swift:64]: What happens if I create a draft from the web, then try to change on mobile ?
(does the web wraps the signature in the same div as we do ?)

File: MailCore/Cache/MailboxManager.swift [hunk 4]
```diff
@@ -79,7 +79,14 @@ public class MailboxManager: ObservableObject {
         let realmName = "\(mailbox.userId)-\(mailbox.mailboxId).realm"
         realmConfiguration = Realm.Configuration(
             fileURL: MailboxManager.constants.rootDocumentsURL.appendingPathComponent(realmName),
-            schemaVersion: 16,
+            schemaVersion: 17,
+            migrationBlock: { migration, oldSchemaVersion in
+                // No migration needed from 0 to 16
+                if oldSchemaVersion < 17 {
```
Comment (adrien-coye) [inline MailCore/Cache/MailboxManager.swift:85]: True that we need to have support for migration now.

**Review by adrien-coye — COMMENTED** [review]
I did a first review.

File: MailCore/Cache/DraftContentManager.swift (hunk 3 above)
Comment (valentinperignon) [reply MailCore/Cache/DraftContentManager.swift:64]: Tested, it works

Comment (adrien-coye) [reply MailCore/Cache/DraftContentManager.swift:64]: Discussed IRL, it works right now as the Web uses the same identifier as we do. All good.

**Review by adrien-coye — APPROVED** [review]
All good, thanks for the changes.

**Pull request merged**